pkg/services/oauthserver/oasimpl/service.go

```diff
@@ -271,6 +271,32 @@ func (s *OAuth2ServiceImpl) SaveExternalService(ctx context.Context, registratio
 	return dto, nil
 }
 
+// SavePluginExternalService is a simplified wrapper around SaveExternalService for the plugin use case.
+func (s *OAuth2ServiceImpl) SavePluginExternalService(ctx context.Context, svcName string, svc *oauth.PluginExternalService) (*oauth.PluginExternalServiceRegistration, error) {
```

Comment:
Nit. I would suggest implementing this in pkg/services/pluginsintegration instead since it's plugin specific and that in turn could call `oauthserver.OAuth2Server.SaveExternalService`.